Source/WebCore/animation/KeyframeEffect.cpp

```diff
@@ -275,6 +275,8 @@ static inline ExceptionOr<KeyframeEffect::KeyframeLikeObject> processKeyframeLik
         // 4. Calculate the normalized property name as the result of applying the IDL attribute name to animation property name algorithm to property name.
         auto propertyName = animationProperties[i].string();
         auto cssPropertyID = IDLAttributeNameToAnimationPropertyName(propertyName);
+        if (!isCSSPropertyExposed(cssPropertyID, &document.settings()))
+            cssPropertyID = CSSPropertyInvalid;
```

Comment:
There's another call site for `IDLAttributeNameToAnimationPropertyName`, I wonder if we should move that code inside, and make `IDLAttributeNameToAnimationPropertyName` take `Settings` as parameter

Reply:
Yeah, I did consider that as a possibility; some of the problem is `IDLAttributeNameToAnimationPropertyName` has a variety of early returns which means moving `isCSSPropertyExposed` to there seems a bigger change which is also inconsistent with everywhere else we convert from strings to CSSPropertyID, not least because in some cases we're nowhere near a Settings object.

Reply:
Maybe the correct thing is to move this check to the call, and then add an ASSERT for the second?

Reply:
we have the CSSPropertyID as return value of `isCSSPropertyExposed` right? can't we store it inside a variable and do the check at the end?